app/models/potion.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class Potion < ApplicationRecord
+  belongs_to :game
+  belongs_to :canonical_potion, optional: true, class_name: 'Canonical::Potion'
+
+  has_many :potions_alchemical_properties, dependent: :destroy, inverse_of: :potion
+  has_many :alchemical_properties, through: :potions_alchemical_properties
+
+  validates :name, presence: true
+  validates :unit_weight, numericality: { greater_than_or_equal_to: 0, allow_blank: true }
+
+  before_validation :set_canonical_potion
+
+  def canonical_models
+    return Canonical::Potion.where(id: canonical_potion_id) if canonical_potion.present?
+
+    matching = Canonical::Potion.where('name ILIKE ?', name)
+    matching = matching.where('magical_effects ILIKE ?', magical_effects) if magical_effects.present?
+    matching = matching.where(**attributes_to_match) if attributes_to_match.any?
```

Comment:
This is intended to provide consistency with other in-game item classes. However, the only `attributes_to_match` value is `unit_weight`, which is 0.5 for all potions. This could make for an extraneous database query. Since this method is different from the corresponding method in other in-game item classes anyway (due to the need to match alchemical properties), would it make sense to just remove this line?

Reply:
Note: there are tests that rely on `unit_weight` being matched.